### DISCUSSION TITLE
Ability to specify an epic for bug and story creation

### DISCUSSION THREAD

**Discussion by jag-j** [forcedotcom/git2gus#180] [Announcements]
We are planning to add the ability to specify an epic for bugs and stories created from Git2Gus. This works as follows.

- Additional configuration in .git2gus/config.json to define epics which contains the work item type along with the GUS epic Id. Example config below:
```
 "issueEpicMapping": {
        "USER STORY": "a3QEE000000ABC1234",
        "BUG P1": "a3QEE000000ABC1235",
        "BUG P2": "a3QEE000000ABC1236",
        "BUG P3": "a3QEE000000ABC1237"
   }
```
- With these configurations defined, the new work items created would be added to the respective epics.

**Comment by mshanemc** [comment]
currently, 

```
"issueTypeLabels": {
    "feature": "USER STORY",
```

means that the label "feature" gets assigned to the WI RecordType named "USER STORY"

I'd suggest maybe some way to maintain that ability (unless you've meaning to translate the stuff after `type:` to the record type? and then add epic.

---

I think a more obvious mapping would be

```
[github-label] : {
  recordType: string,
  epic?: string
}
```

example

```
"feature" : {
  "recordType": "USER STORY",
  "epic": "a3QEE000000ABC1234"
}
```

**Reply by jag-j** [reply]
@mshanemc - My bad! I mistyped what I actually had in mind. I just updated my original post. This would be a new config section called "issueEpicMapping", which is optional. This way it doesn't break the existing git2gus config for users. This would have the issue - epic mapping as below.

```
 "issueEpicMapping": {
        "USER STORY": "a3QEE000000ABC1234",
        "BUG P1": "a3QEE000000ABC1235",
        "BUG P2": "a3QEE000000ABC1236",
        "BUG P3": "a3QEE000000ABC1237"
   }
```
I have also edited the original post.

**Reply by mshanemc** [reply]
ok, that works, too.
